File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/components/TrendingNearYouView.kt
package com.ecommerceearn.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.zIndex
import coil.compose.AsyncImage
import com.ecommerceearn.app.data.model.Product
import com.ecommerceearn.app.data.remote.NetworkClient
import kotlinx.coroutines.launch

@Composable
fun TrendingNearYouView(
    title: String = "Trending near you",
    subtitle: String = "Discover the top products trending today",
    limit: Int = 10,
    productIds: List<String> = emptyList(),
    onProductClick: (Product) -> Unit = {}
) {
    var products by remember { mutableStateOf<List<Product>>(emptyList()) }
    var isLoading by remember { mutableStateOf(true) }

    LaunchedEffect(Unit) {
        try {
            // Replicating simplified fetch logic
            val fetched = NetworkClient.apiService.getProducts(limit)
            products = fetched
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            isLoading = false
        }
    }

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        safeParseColor("#E0FAEF"),
                        safeParseColor("#ECFEFF"),
                        safeParseColor("#F0FDF4")
                    )
                )
            )
            .padding(vertical = 24.dp)
    ) {
        // Header
        Column(modifier = Modifier.padding(horizontal = 16.dp)) {
            Text(
                text = title,
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = safeParseColor("#0D9488")
            )
            if (subtitle.isNotEmpty()) {
                Text(
                    text = subtitle,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    color = safeParseColor("#115E59").copy(alpha = 0.9f)
                )
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        if (isLoading) {
            Box(modifier = Modifier.fillMaxWidth().height(240.dp), contentAlignment = Alignment.Center) {
                CircularProgressIndicator()
            }
        } else if (products.isNotEmpty()) {
            LazyRow(
                contentPadding = PaddingValues(horizontal = 16.dp),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                items(products) { product ->
                    TrendingProductCard(product, onClick = { onProductClick(product) })
                }
            }
        } else {
             Text(
                text = "No trending products found",
                fontSize = 12.sp,
                color = Color.Gray,
                modifier = Modifier.padding(horizontal = 16.dp)
            )
        }
    }
}

@Composable
fun TrendingProductCard(product: Product, onClick: () -> Unit = {}) {
    Column(
        modifier = Modifier
            .width(160.dp)
            .shadow(8.dp, RoundedCornerShape(20.dp))
            .background(Color.White, RoundedCornerShape(20.dp))
            .clickable { onClick() }
            .padding(bottom = 12.dp)
    ) {
        // Image Area
        Box(
            modifier = Modifier.fillMaxWidth()
        ) {
            AsyncImage(
                model = product.images.firstOrNull(),
                contentDescription = null,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(150.dp)
                    .padding(12.dp),
                contentScale = ContentScale.Fit
            )
            
            // Add Button (Overlapping)
            Button(
                onClick = { /* Add */ },
                colors = ButtonDefaults.buttonColors(containerColor = Color.White),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(horizontal = 20.dp, vertical = 6.dp),
                elevation = ButtonDefaults.buttonElevation(defaultElevation = 2.dp),
                modifier = Modifier
                    .align(Alignment.BottomEnd)
                    .offset(x = (-12).dp, y = 14.dp)
                    .zIndex(1f)
                    .height(30.dp)
            ) {
                Text(
                    text = "ADD",
                    color = safeParseColor("#15803D"),
                    fontSize = 13.sp,
                    fontWeight = FontWeight.Bold
                )
            }
        }
        
        Spacer(modifier = Modifier.height(12.dp))

        // Details
        Column(modifier = Modifier.padding(horizontal = 12.dp)) {
            // Weight Subtitle
            if (!product.subtitle.isNullOrEmpty()) {
                Text(
                    text = product.subtitle,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    color = safeParseColor("#6B7280"),
                    modifier = Modifier
                        .background(safeParseColor("#F7F9FC"), RoundedCornerShape(4.dp))
                        .padding(horizontal = 6.dp, vertical = 3.dp)
                )
            }
            
            Spacer(modifier = Modifier.height(4.dp))

            Text(
                text = product.displayName,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                color = safeParseColor("#1F2937"),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
                modifier = Modifier.height(38.dp)
            )

            // Simplistic Price
             Row(
               verticalAlignment = Alignment.Bottom,
               modifier = Modifier.padding(top = 4.dp)
            ) {
               Text(
                   text = "₹${product.price.toInt()}",
                   fontSize = 14.sp,
                   fontWeight = FontWeight.Bold,
                   color = safeParseColor("#1F2937")
               )
               if ((product.mrp ?: 0.0) > product.price) {
                   Spacer(modifier = Modifier.width(4.dp))
                   Text(
                       text = "MRP ₹${product.mrp?.toInt()}",
                       fontSize = 11.sp,
                       textDecoration = TextDecoration.LineThrough,
                       color = safeParseColor("#9CA3AF")
                   )
               }
            }
        }
    }
}

File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/TrendingNearYouView.swift
import SwiftUI

struct TrendingNearYouView: View {
    @State private var products: [Product] = []
    @State private var isLoading = true

    // Props
    var title: String = "Trending near you"
    var subtitle: String = "Discover the top products trending today"
    var limit: Int = 10
    var productIds: [String] = []

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            // Header
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(Color(hex: "#0D9488"))  // Teal 600

                if !subtitle.isEmpty {
                    Text(subtitle)
                        .font(.system(size: 14, weight: .medium))
                        .foregroundColor(Color(hex: "#115E59").opacity(0.9))  // Teal 800
                }
            }
            .padding(.horizontal, 16)

            // Horizontal List
            if isLoading {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 16) {
                        ForEach(0..<3) { _ in
                            RoundedRectangle(cornerRadius: 12)
                                .fill(Color.gray.opacity(0.1))
                                .frame(width: 160, height: 240)
                        }
                    }
                    .padding(.horizontal, 16)
                }
            } else if !products.isEmpty {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 16) {
                        ForEach(products) { product in
                            TrendingProductCard(product: product)
                        }
                    }
                    .padding(.horizontal, 16)
                    .padding(.bottom, 16)  // For shadow
                }
            } else {
                Text("No trending products found")
                    .font(.caption)
                    .foregroundColor(.secondary)
                    .padding(.horizontal, 16)
            }
        }
        .padding(.vertical, 24)
        .background(
            LinearGradient(
                colors: [
                    Color(hex: "#E0FAEF"),
                    Color(hex: "#ECFEFF"),
                    Color(hex: "#F0FDF4"),
                ],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .task {
            await loadProducts()
        }
    }

    private func loadProducts() async {
        do {
            // If explicit IDs provided, typically we'd fetch those.
            // For now, we reuse fetchProducts with a limit or similar logic.
            // Ideally APIService should support `ids` param.
            let fetched = try await APIService.shared.fetchProducts(limit: limit)

            if !productIds.isEmpty {
                // Filter client side if API doesn't support ids param yet,
                // or just take the fetched ones if we can't filter easily consistent with RN logic
                self.products = fetched.filter { productIds.contains($0.id) }
                if self.products.isEmpty {
                    // Fallback if filter leaves empty (e.g. mock ids)
                    self.products = fetched
                }
            } else {
                self.products = fetched
            }
        } catch {
            AppLogger.debug("Failed to load trending products: \(error)")
        }
        isLoading = false
    }
}

struct TrendingProductCard: View {
    let product: Product

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Image
            ZStack(alignment: .bottomTrailing) {
                if let imageUrl = product.images.first, let url = URL(string: imageUrl) {
                    AsyncImage(url: url) { image in
                        image.resizable()
                            .aspectRatio(contentMode: .fit)
                    } placeholder: {
                        Color.gray.opacity(0.1)
                    }
                    .frame(height: 150)
                    .frame(maxWidth: .infinity)
                    .padding(12)
                } else {
                    Rectangle()
                        .fill(Color.gray.opacity(0.1))
                        .frame(height: 150)
                }

                // Add Button
                Button(action: {}) {
                    Text("ADD")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(Color(hex: "#15803D"))
                        .padding(.vertical, 6)
                        .padding(.horizontal, 20)
                        .background(Color.white)
                        .cornerRadius(8)
                        .overlay(
                            RoundedRectangle(cornerRadius: 8)
                                .stroke(Color(hex: "#4D8B4D"), lineWidth: 1)
                        )
                        .shadow(color: .black.opacity(0.1), radius: 2, y: 2)
                }
                .offset(x: -12, y: 14)  // Positioning
            }
            .zIndex(1)  // Keep button on top

            // Details
            VStack(alignment: .leading, spacing: 6) {
                Spacer().frame(height: 8)  // Space for button overlap

                // Weight Badge / Mock removal
                if let subtitle = product.subtitle, !subtitle.isEmpty {
                    Text(subtitle)
                        .font(.system(size: 10, weight: .bold))
                        .foregroundColor(Color(hex: "#6B7280"))
                        .padding(.horizontal, 6)
                        .padding(.vertical, 3)
                        .background(Color(hex: "#F7F9FC"))
                        .cornerRadius(4)
                }

                Text(product.name)
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundColor(Color(hex: "#1F2937"))
                    .lineLimit(2)
                    .frame(height: 38, alignment: .topLeading)

                // Rating
                if let rating = product.rating {
                    HStack(spacing: 2) {
                        ForEach(0..<5) { i in
                            Image(systemName: "star.fill")
                                .font(.system(size: 10))
                                .foregroundColor(i < Int(rating) ? .yellow : .gray.opacity(0.3))
                        }
                        if let count = product.reviewCount {
                            Text("(\(count))")
                                .font(.system(size: 10))
                                .foregroundColor(.gray)
                        }
                    }
                }

                // Delivery - Removed Mock "9 MINS"

                // Price
                HStack(alignment: .bottom, spacing: 6) {
                    Text("₹\(Int(product.price))")
                        .font(.system(size: 14, weight: .bold))
                        .foregroundColor(Color(hex: "#1F2937"))

                    if let mrp = product.mrp, mrp > product.price {
                        Text("MRP ₹\(Int(mrp))")
                            .font(.system(size: 11))
                            .strikethrough()
                            .foregroundColor(Color(hex: "#9CA3AF"))
                    }
                }
            }
            .padding(12)
        }
        .background(Color.white)
        .cornerRadius(20)
        .shadow(color: .black.opacity(0.08), radius: 8, y: 4)
        .frame(width: 160)
    }
}
